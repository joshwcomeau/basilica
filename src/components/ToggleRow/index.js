// eslint-disable-next-line no-unused-vars
import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';

import { capitalize } from '../../utils/misc.utils';
import './index.scss';


const ToggleRow = ({ items, activeItem, onClickItem }) => {
  const classes = classNames(['toggle-row']);

  const itemElements = items.map((item) => {
    const itemClasses = classNames({
      'is-active': item === activeItem,
    });

    return (
      <li className={itemClasses} key={item}>
        {/* Adding a <button> to be a11y-compliant */}
        <button onClick={() => onClickItem(item)}>
          {capitalize(item)}
        </button>
      </li>
    );
  });

  return (
    <div className={classes}>
      <ul>
        {itemElements}
      </ul>
    </div>
  );
};

ToggleRow.propTypes = {
  items: PropTypes.arrayOf(PropTypes.string),
  activeItem: PropTypes.string,
  onClickItem: PropTypes.func,
};

ToggleRow.defaultProps = {

};

export default ToggleRow;
