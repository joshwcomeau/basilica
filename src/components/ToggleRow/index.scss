@import '../../variables';

.toggle-row {
  li {
    position: relative;
    display: inline-block;

    button {
      border: none;
      background: transparent;
      height: 100%;
      padding: 0 90px;
      font-size: 18px;
      font-weight: bold;
      text-transform: uppercase;
      outline: none;
      cursor: pointer;
      color: $grey;
    }

    &:after {
      content: '';
      position: absolute;
      left: 0;
      right: 0;
      bottom: 0;
      height: 3px;
      background: $blue;
      transform-origin: bottom center;
      transform: scaleY(0);
      transition: transform 300ms;
    }

    &.is-active {
      button { color: $black; }

      &:after {
        transform: scaleY(1);
      }
    }
  }

  @media screen and (max-width: $mq-tablet) {

  }

  @media screen and (max-width: $mq-mobile) {

  }
}
