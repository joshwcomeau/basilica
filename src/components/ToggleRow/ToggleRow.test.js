/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import ToggleRow from './index';

describe('ToggleRow', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ToggleRow />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
