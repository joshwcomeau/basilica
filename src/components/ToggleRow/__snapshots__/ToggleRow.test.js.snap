exports[`ToggleRow renders correctly 1`] = `
<div
  className="toggle-row">
  Your Component Here :)
</div>
`;
